[14.0] edi: misc fix/imp

simahawk opened the pull request:
Please, see atomic commits for details.
Most important changes:

* fixed the ordering which was just random due to the override of `_search` CC @etobella 
* added many handy filters and additional menu items
* moved q.job relation from backend to exchange record (see motivation in its commit)

simahawk committed:
edi: fix exc record name compute

No more ugly names in the form view till you populate all fields.

simahawk committed:
edi: improve exchanges views and menu

* add quick filters
* add quick menu items

simahawk committed:
edi: fix exc record missing _rec_name

simahawk committed:
edi: exc record form name not needed

simahawk committed:
edi: fix exc record sorting

simahawk committed:
edi: get rid of name, use identifier

Name contained redundant information like type name and related record.
On for and tree view is totally useless.
To still provide information on linked resource name
the 'related_name' field has been added.

simahawk committed:
edi: backend action to view exchanges

simahawk committed:
edi: backend action to view exchange types

simahawk committed:
edi_storage: fail gracefully w/o error report

simahawk committed:
edi: move job func to exc record

Prior to this change jobs were tied to the backend.

The backend must be responsible for processing records
and this is why the logic has been implemented there
and it's kept there with this change too.

However, it's way better if the job is tied to its record directly
because it makes more sense and it helps jumping to the record
w/ the 'related' btn on the job.

simahawk committed:
edi: update exc type adv settings help

ivantodorovich reviewed (approved):
LG

etobella commented on edi_oca/data/job_function.xml:
```diff
@@ -1,26 +1,26 @@
-        <field name="model_id" ref="model_edi_backend" />
+    <record id="job_fun_exchange_record_generate" model="queue.job.function">
+        <field name="model_id" ref="model_edi_exchange_record" />
         <field name="method">exchange_generate</field>
```
`action_exchange_generate`, isn't it?

simahawk replied:
good catch Quique! Tnx :)

simahawk commented:
test failure seems unrelated
```
2022-03-17 08:08:18,090 400 ERROR odoo odoo.addons.account_invoice_import_invoice2data.tests.test_invoice_import: ERROR: TestInvoiceImport.test_import_free_invoice
Traceback (most recent call last):
  File "/__w/edi/edi/account_invoice_import_invoice2data/wizard/account_invoice_import.py", line 53, in invoice2data_parse_invoice
    invoice2data_res = extract_data(file_name, templates=templates)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/main.py", line 93, in extract_data
    return t.extract(optimized_str)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 204, in extract
    result = parsers.regex.parse(self, {"regex": v, "type": "date"}, optimized_str, True)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/parsers/regex.py", line 36, in parse
    result[k] = template.coerce_type(v, settings["type"])
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 151, in coerce_type
    return self.parse_date(value)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 136, in parse_date
    languages=self.options["languages"],
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/conf.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/__init__.py", line 61, in parse
    data = parser.get_date_data(date_string, date_formats)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 429, in get_date_data
    locale, date_string, date_formats, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 178, in parse
    return instance._parse()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 182, in _parse
    date_data = self._parsers[parser_name]()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 196, in _try_freshness_parser
    return freshness_date_parser.get_date_data(self._get_translated_date(), self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 235, in _get_translated_date
    self.date_string, keep_formatting=False, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 131, in translate
    relative_translations = self._get_relative_translations(settings=settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 158, in _get_relative_translations
    self._generate_relative_translations(normalize=True))
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 172, in _generate_relative_translations
    pattern = DIGIT_GROUP_PATTERN.sub(r'?P<n>\d+', pattern)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/regex.py", line 700, in _compile_replacement_helper
    is_group, items = _compile_replacement(source, pattern, is_unicode)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/_regex_core.py", line 1736, in _compile_replacement
    raise error("bad escape \\%s" % ch, source.string, source.pos)
regex._regex_core.error: bad escape \d at position 7
```

mmequignon commented on edi_oca/models/edi_exchange_record.py:
```diff
@@ -110,11 +110,11 @@ class EDIExchangeRecord(models.Model):
 
-    @api.depends("type_id.code", "model", "res_id")
-    def _compute_name(self):
+    @api.depends("model", "res_id")
```
this depends looks wrong?
```python
@api.depends("record", "record.display_name")
```

simahawk replied:
is not :) `record` is a property computed from res_id and model ;)

mmequignon commented on edi_oca/templates/exchange_chatter_msg.xml:
```diff
@@ -13,7 +13,7 @@
                     t-att-data-oe-id="exchange_record.id"
                 >
-                    <span t-field="exchange_record.name" />
+                    <span t-field="exchange_record.identifier" />
```
why identifier ? I was expecting `related_name`

simahawk replied:
this msg is posted on the related record to say that it has been created or update BY the exc record

mmequignon reviewed (approved):
LG, just tiny questions

simahawk commented:
Build fixed here https://github.com/OCA/edi/pull/547

simahawk committed:
edi: show exchanges first

Makes little sense to show backends first
when you enter the EDI menu.
Ongoing exchanges are the most important thing.

simahawk committed:
edi_oca: avoid warning on exception args

simahawk commented:
/ocabot merge minor

etobella commented:
@simahawk removal of some functions is giving problems to other areas :sob: https://github.com/OCA/l10n-spain/runs/5657494643?check_suite_focus=true

simahawk commented:
@etobella ah because you are using it directly? I see, well, we can restore them and deprecate them... sorry, I didn't know :sweat_smile: .

What do you prefer? Adapt your implementation or adding back and depracating?

simahawk commented:
I'd prefer "adapting" but just because it's cleaner :stuck_out_tongue_winking_eye:

etobella commented:
I would prefer adding them, because it happens to me, but it could happen somewhere else. Just marking them as deprecated and removal on 16 :wink:
